tsconfig.json

```diff
@@ -14,6 +14,7 @@
   "include": [
     "packages/core/src/util",
     "packages/core/src/tracing/*.js",
+    "packages/core/src/tracing/instrumentation/protocols/_http.js",
```

Comment:
question: Is `_http` exposed as part of the collector package? What was the reason to add TS to it?

Reply:
No, it is not exposed. The reason is two-fold. I noticed that a lot of the method in that file didn't have any doc comment at all, and I wanted to fix that. There are a lot of very similar, yet slightly different methods in there -- all focussed on capturing headers from HTTP requests, tailored to very specific situations (capture from a Request object, from an object literal, merge headers etc.). It took me a while to sort out which method is doing what exactly, so I figured I should just add doc comments to them.

And while I added comments to the methods, I figured it would be best to use the JSDoc comment format, which we have in a lot of places already and which describes parameters and return values in a standardized format. Having done that, I also wanted to make sure I declared the correct types for the parameters, so that's why I added this to the typescript files. 

And the changes that were necessary in _http.js for that (except for the method comments) are rather small, just a three or four inline type declarations that's all. Added bonus: Adding types uncovered a couple of edge cases in handling the headers that we might never have noticed otherwise.

Long story short, it escalated from adding a few comments to adding types and having tsc check them.

**I can absolutely remove the declaration here if you prefer that.** In my opinion, having utility files like this type check is nice, but I do not hold that opinion strongly.

Reply:
👍 Lets keep the ts types